### ISSUE TITLE
Bug: `no-underscore-dangle` not working as expected in ES2022 public/private class fields

### ISSUE THREAD

**Issue by robertotcestari** [eslint/eslint#15810]
### Environment

Node version: 16.13
npm version: 8.1.2
Local ESLint version: 8.14.0
Global ESLint version: - 
Operating System: macOS m1 pro

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "extends": "eslint:recommended",
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "parserOptions": {
    "ecmaVersion": 2022
  },
  "rules": {
    "no-underscore-dangle": [
      2,
      {
        "enforceInMethodNames": true
      }
    ]
  }
}

```
</details>

### What did you expect to happen?

The rule `no-underscore-dangle` should also apply to the new ES2022 public/private class fields.
The implementation was marked _done_ https://github.com/eslint/eslint/issues/14857 , however, _it still does not work_, as no errors are shown in public/private class fields. 

### What actually happened?

Errors were raised only for `methods` in the class (and properties in the constructor), but not for methods or properties defined via `public/private class fields` syntax. 

<img width="846" alt="image" src="https://user-images.githubusercontent.com/14261421/165385930-28e1e5db-e453-4ad0-9401-c6fcb8df76d8.png">

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by btmills** [comment]
I was able to repro this [in the demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVuZGVyc2NvcmUtZGFuZ2xlOiBbXCJlcnJvclwiLCB7IFwiZW5mb3JjZUluTWV0aG9kTmFtZXNcIjogdHJ1ZSB9XSAqL1xuXG5jbGFzcyBDbGFzcyB7XG4gIF9wdWJsaWNGaWVsZCA9IFwiZmFsc2UgbmVnYXRpdmVcIjtcbiAgI19wcml2YXRlRmllbGQgPSBcImZhbHNlIG5lZ2F0aXZlXCI7XG4gIFxuICBfcHVibGljTWV0aG9kKCkgeyByZXR1cm4gXCJ0cnVlIHBvc2l0aXZlXCI7IH1cbiAgI19wcml2YXRlTWV0aG9kKCkgeyByZXR1cm4gXCJ0cnVlIHBvc2l0aXZlXCI7IH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICB0aGlzLl9wdWJsaWNGaWVsZCA9IFwidHJ1ZSBwb3NpdGl2ZVwiO1xuICAgIHRoaXMuI19wcml2YXRlRmllbGQgPSBcInRydWUgcG9zaXRpdmVcIjtcbiAgfVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) (or the [new demo](https://play.eslint.org/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVuZGVyc2NvcmUtZGFuZ2xlOiBbXCJlcnJvclwiLCB7IFwiZW5mb3JjZUluTWV0aG9kTmFtZXNcIjogdHJ1ZSB9XSAqL1xuXG5jbGFzcyBDbGFzcyB7XG4gIF9wdWJsaWNGaWVsZCA9IFwiZmFsc2UgbmVnYXRpdmVcIjtcbiAgI19wcml2YXRlRmllbGQgPSBcImZhbHNlIG5lZ2F0aXZlXCI7XG4gIFxuICBfcHVibGljTWV0aG9kKCkgeyByZXR1cm4gXCJ0cnVlIHBvc2l0aXZlXCI7IH1cbiAgI19wcml2YXRlTWV0aG9kKCkgeyByZXR1cm4gXCJ0cnVlIHBvc2l0aXZlXCI7IH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICB0aGlzLl9wdWJsaWNGaWVsZCA9IFwidHJ1ZSBwb3NpdGl2ZVwiO1xuICAgIHRoaXMuI19wcml2YXRlRmllbGQgPSBcInRydWUgcG9zaXRpdmVcIlxuICB9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6eyJlczYiOnRydWV9fX0=)!).

```js
/* eslint no-underscore-dangle: ["error", { "enforceInMethodNames": true }] */

class Class {
  _publicField = "false negative";
  #_privateField = "false negative";
  
  _publicMethod() { return "true positive"; }
  #_privateMethod() { return "true positive"; }

  constructor() {
    this._publicField = "true positive";
    this.#_privateField = "true positive";
  }
}
```

In my opinion, class fields are new enough that we could fix this in a semver-minor release. If the team disagrees, the alternative is putting this check behind an option just like `enforceInMethodNames`, so potentially `enforceInClassFields`, which defaults to false to maintain compatibility with the current behavior and then default it to true in the next major release.

**Comment by robertotcestari** [comment]
The option `enforceInClassFields` is a great suggestion, @btmills . There is this https://github.com/eslint/eslint/pull/15811, but i am happy to submit another code with the `enforceInClassFields` option property.

**Comment by mdjermanovic** [comment]
Since this is a stylistic rule, and it already doesn't check all identifiers (for example, it allows property names in object literals), I think we should add this new check behind an option to maintain compatibility with the current behavior.

`enforceInClassFields` (default `false`) sounds good to me.

**Comment by robertotcestari** [comment]
Hi @btmills @mdjermanovic, just submitted a new PR (https://github.com/eslint/eslint/pull/15818) with

- New option `enforceInClassFields` (default `false`)
- Test cases
- Changes in documentation where appropriate. 

This should fix and address your suggestions.
